maven-core/src/main/java/org/apache/maven/graph/DefaultGraphBuilder.java

```diff
@@ -138,8 +138,10 @@ public DefaultGraphBuilder( BuildResumptionDataRepository buildResumptionDataRep
     {
         ProjectDependencyGraph projectDependencyGraph = new DefaultProjectDependencyGraph( projects );
         List<MavenProject> activeProjects = projectDependencyGraph.getSortedProjects();
+        List<MavenProject> allSortedProjects = projectDependencyGraph.getSortedProjects();
```

Comment:
It is the same list as above. It it correct?

Reply:
It starts like that, but `activeProjects` is constantly being overwritten each step in the reactor the activeProjects list. This is a pattern I don't like (I prefer immutable variables). It's one of the things I fixed in the bigger refactoring.